src/java/org/apache/cassandra/repair/SystemDistributedKeyspace.java

```diff
@@ -307,6 +320,55 @@ public static void setViewRemoved(String keyspaceName, String viewName)
         forceBlockingFlush(VIEW_BUILD_STATUS);
     }
 
+    /**
+     * Reads blacklisted partitions from system_distributed.blacklisted_partitions table.
+     * Stops reading partitions upon exceeding the cache size limit by logging a warning.
+     * @return
+     */
+    public static Set<BlacklistedPartition> getBlacklistedPartitions()
+    {
+        String query = "SELECT keyspace_name, columnfamily_name, partition_key FROM %s.%s";
+        UntypedResultSet results;
+        try
+        {
+            results = QueryProcessor.execute(format(query, SchemaConstants.DISTRIBUTED_KEYSPACE_NAME, BLACKLISTED_PARTITIONS),
+                                             ConsistencyLevel.ONE);
+        }
+        catch (Exception e)
+        {
+            logger.error("Error querying blacklisted partitions", e);
+            return Collections.emptySet();
+        }
+
+        Set<BlacklistedPartition> blacklistedPartitions = new HashSet<>();
+        int cacheSize = 0;
```

Comment:
nit: might be easier to refer to this with a unit so that the divide by later makes sense e.g. `cacheSizeBytes`

Reply:
done